### PR TITLE
Form

### PR DESCRIPTION
**Pull request by SaeqaSultani**
- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
1- It will return the first user from the database.
2- Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
1- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
2- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/post_controller.rb [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+    @comments = @post.comments
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
```
Comment (fobadara) [inline app/controllers/post_controller.rb:19]: [OPTIONAL] You could have done this:
```
      @post = @current_user.posts.new(post_params)

```

File: app/controllers/comment_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      redirect_to user_post_index_path(current_user, @current_post)
+
+    else
+      flash[:error] = 'Something went wrong'
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def comment_params
+    params.permit(:text)
```
Comment (fobadara) [inline app/controllers/comment_controller.rb:22]: [OPTIONAL] Kindly restrict parameters like this:
`params.require(:comment).permit(:text)`

File: config/routes.rb [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
   # Defines the root path route ("/")
-  # root "articles#index"
+  root 'user#index'
   resources :user, only: [:show,:index] do    
-    resources :post, only: [:index, :show]     
-       end
+    resources :post, only: [:index, :show, :new, :create] do
+      resources :comment, only: [:create]
```
Comment (fobadara) [inline config/routes.rb:9]: [REQUIRED] Kindly include routes for the `new` comment form

File: app/controllers/like_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class LikeController < ApplicationController
+  def new
+    @like = Like.new
+  end
+
+  def create
+    @current_post = Post.find(params[:post_id])
+    @like = @current_post.likes.new
+    @like.post_id = @current_post.id
+    @like.author_id = current_user.id
```
Comment (fobadara) [inline app/controllers/like_controller.rb:10]: [OPTIONAL] You could have just done this:

```
 @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)
```

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @SaeqaSultani, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Good first attempt
✅ Project is well documented

## Required Changes ♻️
_Check the comments under the review._
I had to include the comment here because there is no file to make inline comments on
[REQUIRED] I can't find a views/comments/new.html.erb file for new comments.
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @SaeqaSultani,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job Implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**